frontend/src/components/Editor.vue

```diff
@@ -1,79 +1,68 @@
 <script setup lang="ts">
+/*
+ * Component used to display code mirror editor
+ */
+
+import { ref, watch, onUnmounted, defineModel, markRaw, PropType } from 'vue';
+
+import CodeMirror from 'codemirror';
+import { lintPipeline } from '../PipelineValidation.js';
+import { curTheme as curThemeSvelte } from '../utils.js';
+import 'codemirror/lib/codemirror.css';
+import 'codemirror/mode/clike/clike.js';
+import 'codemirror/mode/yaml/yaml.js';
+import 'codemirror/mode/python/python.js';
+import 'codemirror/mode/markdown/markdown.js';
+import 'codemirror/mode/htmlmixed/htmlmixed.js';
+import 'codemirror/addon/display/fullscreen.js';
+import 'codemirror/addon/display/fullscreen.css';
+import 'codemirror/addon/lint/lint.js';
+import 'codemirror/addon/lint/lint.css';
+import 'codemirror/addon/hint/show-hint.js';
+import 'codemirror/addon/hint/show-hint.css';
+import { useReadableSvelteStore } from '../utilities/useSvelteStoreInVue';
+
+/**
+ * @prop {string}  filename   - used to determine hints and highlighting from extension, default ''
+ * @prop {boolean} autofocus  - enable/disable autofocus on editor, default false
+ * @prop {boolean} disabled   - if true then editor is read only, default false
+ * @prop {boolean} wrap       - wrap long lines, default false
+ * @prop {Record<string, () => void>} extraKeys
+ * default keys are:
+ *    - Ctrl-Space   autocomplete
+ *    - F11          full screen
+ *    - ESC          escape full screen
+ *    - Tab          soft tab
+ *    You can add custom ones in form { 'key': function }
+ */
+const props = defineProps({
```

Comment:
We use the type-based `defineProps` approach everywhere else in Kelvin (`defineProps<{...}>(...)`). Was there something wrong with the original `defineProps` call here?